### PR TITLE
Removing retryAfterCount for manual jobs

### PR DESCRIPTION
**Pull request by chiragsalian**
### Fixed Issues
Fixes https://github.com/Expensify/Expensify/issues/220939

### Tests
1. I created two jobs. One manual and another non manual,
```
CreateJob
name: manual/test
retryAfter: +10 minutes
```
```
CreateJob
name: www-prod/test
retryAfter: +10 minutes
```
2. I called GetJobs on them,
```
GetJob
name: manual/test
```
```
GetJob
name: www-prod/test
```
3. I confirmed the manual job did not have its retryAfterCount incremented while the prod job did,
```
sqlite> select * from jobs;
created              jobID                state      name           nextRun              lastRun              repeat  data                   priority  parentJobID  retryAfter
-------------------  -------------------  ---------  -------------  -------------------  -------------------  ------  ---------------------  --------  -----------  -----------
2022-09-19 23:52:31  185975920383287168   RUNQUEUED  manual/test    2022-09-20 00:02:40  2022-09-19 23:52:40          {}                     500       0            +10 minutes
2022-09-19 23:55:33  1689846290350887486  RUNQUEUED  www-prod/test  2022-09-20 00:05:38  2022-09-19 23:55:38          {"retryAfterCount":1}  500       0            +10 minutes
```
_________
**Internal Testing Reminder:** when changing bedrock, please compile auth against your new changes

### CODE REVIEW

File: plugins/Jobs.cpp [hunk 1]
```diff
@@ -842,13 +842,19 @@ void BedrockJobsCommand::process(SQLite& db) {
                 string retryAfterDateTime = "DATETIME(" + SQ(currentTime) + ", " + SQ(job["retryAfter"]) + ")";
                 string repeatDateTime = _constructNextRunDATETIME(job["nextRun"], currentTime, job["repeat"]);
                 string nextRunDateTime = repeatDateTime != "" ? "MIN(" + retryAfterDateTime + ", " + repeatDateTime + ")" : retryAfterDateTime;
+                bool isManualJob = job["name"].rfind("manual", 0) == 0;
```
Comment (flodnv) [inline plugins/Jobs.cpp:845]: Use SStartsWith?

File: plugins/Jobs.cpp [hunk 2]
```diff
@@ -842,13 +842,19 @@ void BedrockJobsCommand::process(SQLite& db) {
...
+                    // We also set originalNextRun so we don't lose track of the original nextRun (which we are overriding here)
+                    dataUpdateQuery = ", data = JSON_SET(data, '$.retryAfterCount', COALESCE(JSON_EXTRACT(data, '$.retryAfterCount'), 0) + 1" + (isRepeatBasedOnScheduledTime ? ", '$.originalNextRun', " + SQ(job["nextRun"]) + ") ": ") ");
+                }
                 string updateQuery = "UPDATE jobs "
                                      "SET state = 'RUNQUEUED', "
                                          "lastRun = " + SQ(currentTime) + ", " +
-                                         "nextRun = " + nextRunDateTime + ", " +
-                                         "data = JSON_SET(data, '$.retryAfterCount', COALESCE(JSON_EXTRACT(data, '$.retryAfterCount'), 0) + 1" + // Set this so we don't retry infinitely (see above)
-                                         (isRepeatBasedOnScheduledTime ? ", '$.originalNextRun', " + SQ(job["nextRun"]) + ") ": ") ") + // Set this so we don't lose track of the original nextRun (which we are overriding here)
+                                         "nextRun = " + nextRunDateTime + "" +
```
Comment (flodnv) [inline plugins/Jobs.cpp:856]:
```suggestion
                                         "nextRun = " + nextRunDateTime +
```

**Review by flodnv — CHANGES_REQUESTED** [review]

**Comment by chiragsalian** [comment]
Updated and ready for review.

**Review by flodnv — APPROVED** [review]
Thanks! @tylerkaraszewski can you do a final pass please? 🙏

**Review by tylerkaraszewski — APPROVED** [review]

**Pull request merged**